In [0]:
import pandas as pd
import matplotlib.pyplot as plt
from pandasql import sqldf
pysqldf = lambda q: sqldf(q, globals())

import import_ipynb
from week1_data_analysis import summer_events, winter_events

importing Jupyter notebook from week1_data_analysis.ipynb


## Más allá de las estadísticas descriptivas

In [0]:
#Summer Olympics:
summer_medal_count = pysqldf('''
                         SELECT
                             Year,
                             COUNT(*) AS total_count,
                             SUM(CASE
                                   WHEN Medal IS NOT NULL THEN 1 ELSE 0
                                 END) AS medal_count,
                             SUM(CASE
                                   WHEN Medal = "Gold" THEN 1 ELSE 0
                                 END) AS gold_count,
                             SUM(CASE
                                   WHEN Medal = "Silver" THEN 1 ELSE 0
                                 END) AS silver_count,
                             SUM(CASE
                                   WHEN Medal = "Bronze" THEN 1 ELSE 0
                                 END) AS bronze_count
                             FROM
                               summer_events
                             GROUP BY
                               Year 
                                   ''')

In [0]:
#Winter Olympics:
winter_medal_count = pysqldf('''
                         SELECT
                             Year,
                             COUNT(*) AS total_count,
                             SUM(CASE
                                   WHEN Medal IS NOT NULL THEN 1 ELSE 0
                                 END) AS medal_count,
                             SUM(CASE
                                   WHEN Medal = "Gold" THEN 1 ELSE 0
                                 END) AS gold_count,
                             SUM(CASE
                                   WHEN Medal = "Silver" THEN 1 ELSE 0
                                 END) AS silver_count,
                             SUM(CASE
                                   WHEN Medal = "Bronze" THEN 1 ELSE 0
                                 END) AS bronze_count
                             FROM
                               winter_events
                             GROUP BY
                               Year 
                                   ''')

Acabo de crear dos tablas para contar el número total de medallas en los Juegos Olímpicos de invierno y verano. Calcularé el coeficiente de correlación de Pearson entre el número total de medallas en los Juegos Olímpicos de invierno y de verano.

In [0]:
print(summer_medal_count.head())


   Year  total_count  medal_count  gold_count  silver_count  bronze_count
0  1896          380          143          62            43            38
1  1900         1936          604         201           228           175
2  1904         1301          486         173           163           150
3  1906         1733          458         157           156           145
4  1908         3101          831         294           281           256


In [0]:
print(winter_medal_count.head())

   Year  total_count  medal_count  gold_count  silver_count  bronze_count
0  1924          460          130          55            38            37
1  1928          582           89          30            28            31
2  1932          352           92          32            32            28
3  1936          895          108          36            37            35
4  1948         1075          135          41            48            46


In [0]:
print(summer_medal_count)

    Year  total_count  medal_count  gold_count  silver_count  bronze_count
0   1896          380          143          62            43            38
1   1900         1936          604         201           228           175
2   1904         1301          486         173           163           150
3   1906         1733          458         157           156           145
4   1908         3101          831         294           281           256
5   1912         4040          941         326           315           300
6   1920         4292         1308         493           448           367
7   1924         5233          832         277           281           274
8   1928         4992          734         245           239           250
9   1932         2969          647         229           214           204
10  1936         6506          917         312           310           295
11  1948         6405          852         289           284           279
12  1952         8270    

The length of the array of the number of medal count in the winter Olympics and summer Olympics are different because Winter olympics started in 1924, but Summer olympics started in 1896. Therefore I have to create a new shortened table of the summer olympics started in 1924 to match the length of the winter olympics.

In [0]:
summer_medal_count_new = summer_medal_count[7:]

In [0]:
print(summer_medal_count_new)

    Year  total_count  medal_count  gold_count  silver_count  bronze_count
7   1924         5233          832         277           281           274
8   1928         4992          734         245           239           250
9   1932         2969          647         229           214           204
10  1936         6506          917         312           310           295
11  1948         6405          852         289           284           279
12  1952         8270          897         306           291           300
13  1956         5127          893         302           293           298
14  1960         8119          911         309           294           308
15  1964         7702         1029         347           339           343
16  1968         8588         1057         359           340           358
17  1972        10304         1215         404           392           419
18  1976         8641         1320         438           434           448
19  1980         7191    

Luego calculo el coeficiente de correlación de Pearson entre el número total de medallas en los Juegos Olímpicos de invierno y verano desde 1924 hasta 2016.

In [0]:
import numpy as np

In [0]:
x_simple = winter_medal_count.medal_count
y_simple = summer_medal_count_new.medal_count
my_rho = np.corrcoef(x_simple, y_simple)

In [0]:
print(my_rho)

[[1.         0.94141801]
 [0.94141801 1.        ]]


El coeficiente de correlación de Pearson entre el número total de medallas en los Juegos Olímpicos de invierno y verano de 1924 a 2016 es de 0,94, lo que es muy positivo. Por lo tanto, el desempeño de un país en los Juegos Olímpicos de invierno está altamente correlacionado con el de los Juegos Olímpicos de verano.

Luego calcularé la desviación estándar del desempeño del país a lo largo de los años. Será útil una comparación entre el estándar promedio de invierno y el de los Juegos Olímpicos de verano.

In [0]:
std_medal_count_summer_olympics = np.std(y_simple)
std_medal_count_winter_olympics = np.std(x_simple)


In [0]:
print("std_medal_count_summer_olympics =",std_medal_count_summer_olympics)
print("std_medal_count_winter_olympics =",std_medal_count_winter_olympics)

std_medal_count_summer_olympics = 475.323015441357
std_medal_count_winter_olympics = 152.56899942903493


De 1924 a 2016, como la desviación estándar en los Juegos Olímpicos de verano es aproximadamente 3 veces mayor que en los Juegos Olímpicos de invierno, el desempeño de los países por año cambia más en los Juegos Olímpicos de verano.